### PR TITLE
#6665 proposal

### PR DESCRIPTION
**Pull request by tandraschko**

### CODE REVIEW

File: src/main/java/org/primefaces/model/TableState.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ * THE SOFTWARE.
+ */
+package org.primefaces.model;
+
+import java.util.List;
+import java.util.Map;
+import javax.el.MethodExpression;
+import org.primefaces.model.filter.FilterConstraint;
+
+public class TableState {
```
Comment (melloware) [inline src/main/java/org/primefaces/model/TableState.java:31]: Serializable?

File: src/main/java/org/primefaces/model/TableState.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    
+    private int first;
+
+    private int rows;
+    
+    private Map<String, ColumnState> columns;
+    
+    private Object globalFilterValue;
```
Comment (melloware) [inline src/main/java/org/primefaces/model/TableState.java:39]: transient?

**Review by melloware — CHANGES_REQUESTED** [review]
If we were using Lombok you could just slap `@Builder` and `@Data` on these two classes :)

**Comment by tandraschko** [comment]
please dont review in this detail, that is just to show what data i would like to store in which form :D Far away from anything real.

**Comment by tandraschko** [comment]
i really like the structure in general, im just unsure how a user should define default sort/filter via code now...

**Pull request closed without merging**